packages/cli/src/modules/start/commands/package/start/startBackend.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs-extra';
+import { resolve as resolvePath } from 'path';
+import { paths } from '../../../../../lib/paths';
+import { runBackend } from '../../../../../lib/runner';
+
+interface StartBackendOptions {
+  targetDir: string;
+  checksEnabled: boolean;
+  inspectEnabled?: boolean | string;
+  inspectBrkEnabled?: boolean | string;
+  linkedWorkspace?: string;
+  require?: string;
+}
+
+export async function startBackend(options: StartBackendOptions) {
+  const waitForExit = await runBackend({
+    targetDir: options.targetDir,
+    entry: 'src/index',
+    inspectEnabled: options.inspectEnabled,
+    inspectBrkEnabled: options.inspectBrkEnabled,
+    linkedWorkspace: options.linkedWorkspace,
+    require: options.require,
+  });
+
+  await waitForExit();
+}
+
+export async function startBackendPlugin(options: StartBackendOptions) {
+  const hasDevIndexEntry = await fs.pathExists(
+    resolvePath(options.targetDir ?? paths.targetDir, 'dev/index.ts'),
+  );
+  if (!hasDevIndexEntry) {
+    console.warn(
```

Comment:
is it really ok to just bail with a "happy" error code here?

Reply:
this is existing code, did have the same thought though, thinking it's fine as is for now